persistence/internal/providertest/offsetstore/repository.go

```diff
@@ -0,0 +1,70 @@
+package offsetstore
+
+import (
+	"context"
+
+	"github.com/dogmatiq/infix/persistence"
+	"github.com/dogmatiq/infix/persistence/internal/providertest/common"
+	"github.com/dogmatiq/infix/persistence/subsystem/offsetstore"
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+)
+
+// DeclareRepositoryTests declares a functional test-suite for a specific
+// offsetstore.Repository implementation.
+func DeclareRepositoryTests(tc *common.TestContext) {
+	ginkgo.Describe("type offsetstore.Repository", func() {
+		var (
+			dataStore  persistence.DataStore
+			repository offsetstore.Repository
+			tearDown   func()
+		)
+
+		ginkgo.BeforeEach(func() {
+			dataStore, tearDown = tc.SetupDataStore()
+			repository = dataStore.OffsetStoreRepository()
+		})
+
+		ginkgo.AfterEach(func() {
+			tearDown()
+		})
+
+		ginkgo.Describe("func LoadOffset()", func() {
+			ginkgo.When("application has no previous offsets associated", func() {
+				ginkgo.It("loads the initial offset as zero", func() {
+					actual, err := repository.LoadOffset(
+						context.Background(),
```

Comment:
```suggestion
						tc.Context,
```

The test context has a `context.Context` in it that handles timing out the tests, etc.

Reply:
Oh you've used it elsewhere, just this one.